src/composables/transactionItemList.ts

```diff
@@ -0,0 +1,128 @@
+import { computed } from '@vue/composition-api';
+import {
+  IActiveMultisigTx,
+  IDefaultComposableOptions,
+  ITokenList,
+  ITransaction,
+  ITx,
+} from '../types';
+import {
+  AETERNITY_CONTRACT_ID,
+  defaultTransactionSortingCallback,
+  getInnerTransaction,
+  TX_FUNCTIONS,
+} from '../popup/utils';
+import { useTransactionTx } from './transactionTx';
+import { usePendingMultisigTransaction } from './pendingMultisigTransaction';
+import { useTransactionAndTokenFilter } from './transactionAndTokenFilter';
+import { i18n } from '../store/plugins/languages';
+
+interface TransactionItemList extends IDefaultComposableOptions {
+  isMultisig?: boolean,
+  token: string,
+}
+
+type ITransactionElement = ITransaction | IActiveMultisigTx
+
+export const useTransactionItemList = ({
+  store,
+  isMultisig,
+  token,
+}: TransactionItemList) => {
+  const availableTokens = computed<ITokenList>(() => store.state.fungibleTokens.availableTokens);
+  const transactions = computed(() => store.state.transactions);
+  const getTxSymbol = computed(() => store.getters.getTxSymbol);
+  const getAccountPendingTransactions = computed<IActiveMultisigTx[]>(
+    () => store.getters.getAccountPendingTransactions,
+  );
+
+  const canLoadMore = computed(() => !!transactions.value.nextPageUrl);
+
+  const {
+    searchPhrase,
+    displayMode,
+    FILTER_MODE,
+  } = useTransactionAndTokenFilter();
+
+  const { pendingMultisigTransaction } = usePendingMultisigTransaction({ store });
+
+  function isFungibleTokenTx(tx: ITx) {
+    return Object.keys(availableTokens.value).includes(tx.contractId);
+  }
+
+  function filterProperTransactions(tr: ITransactionElement) {
+    const innerTx = getInnerTransaction(tr.tx!);
+    return !token || (
+      token !== AETERNITY_CONTRACT_ID
+        ? innerTx?.contractId === token
+        : (!innerTx.contractId || !isFungibleTokenTx(innerTx))
+    );
+  }
+
+  function filterTransactionByDisplayMode(tr: ITransactionElement) {
+    const { direction, isDex } = useTransactionTx({
+      store,
+      tx: tr.tx,
+      externalAddress: (tr as ITransaction).transactionOwner,
+    });
+
+    switch (displayMode.value.key) {
+      case FILTER_MODE.all:
+        return true;
+      case FILTER_MODE.dex:
+        return isDex.value;
+      case FILTER_MODE.out:
+        return direction.value === TX_FUNCTIONS.sent && !isDex.value;
+      case FILTER_MODE.in:
+        return direction.value === TX_FUNCTIONS.received;
+      default:
+        throw new Error(`${i18n.t('pages.recentTransactions.unknownMode')} ${displayMode.value.key}`);
+    }
+  }
+
+  function filterTransactionByPhrase(tr: ITransactionElement) {
+    return !searchPhrase.value || getTxSymbol.value(tr)
+      .toLocaleLowerCase()
+      .includes(searchPhrase.value.toLocaleLowerCase());
+  }
+
+  const loadedTransaction = computed((): ITransactionElement[] => {
```

Comment:
As you return multiple transactions
```suggestion
  const loadedTransactionList = computed((): ITransactionElement[] => {
```